src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
-    @Override
-    public void add(T value) {
+    private static final int ZERO_SIZE = 0;
+    private static final int HALF_SIZE = 2;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T data;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(T data) {
+            this.data = data;
+        }
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T element) {
+        Node<T> newNode = new Node<>(element);
+        if (tail == null) {
+            head = tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
-    public void addAll(List<T> list) {
+    public void add(T element, int index) {
+        if (index < ZERO_SIZE || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index
+                    + ", Size: " + size);
+        }
+        Node<T> newNode = new Node<>(element);
+        if (index == size) {
+            add(element);
+        } else if (index == ZERO_SIZE) {
```

Comment:
Don't complicate if-else construction. Consider refactoring the add method to avoid repetition and improve readability.